main/src/library/Concurrent/Channel.flix

```diff
@@ -420,14 +435,15 @@ namespace Concurrent/Channel {
         MutList.clear!(waitingGetters)
 
     ///
-    /// Signals and clears the waiting putters.
+    /// Either signals waiting putters or rendezvous as appropriate.
     /// The channel lock is expected to be held.
     ///
-    def signalPutters(c: MpmcAdmin): Unit \ IO =
-        let MpmcAdmin(_, _, _, _, _, _, waitingPutters) = c;
-        // Signal waiting setters that the channel has space
-        signalCondition(waitingPutters)
-        // Since there is only one condition there is no cleaning up to do.
+    def onGetComplete(c: MpmcAdmin): Unit \ IO =
+        let MpmcAdmin(_, _, unBuffered, _, _, _, rendezvous, waitingPutters) = c;
+        if (unBuffered)
+            signalCondition(rendezvous)
+        else 
+            signalCondition(waitingPutters)
```

Comment:
Why is this not done always? `put` will call `awaitCondition(waitingPutters)` and to me it doesn't seem like anything wakes it up in the unbuffered case

Reply:
So the buffered case works exactly as before: the `rendezvous` condition is completely unused in that case.

For the buffered case we have to proceed in two steps:

1. First we unblock the thread which successfully sent a value (the value that was just received)
2. Then we unblock `waitingPutters`.

This second step is performed by the successful putter [here](https://github.com/flix/flix/blob/61eeb02cb756c0a881c4acd6fb868eca06fc65e6/main/src/library/Concurrent/Channel.flix#L175)

I think this two step approach is unavoidable: we need to be sure that the single thread waiting on `rendezvous` (the one that wrote the value that's just been read) is woken up first, or the danger is that another waiting putter will be woken up first, write a new value, and now `size` is 1.